grakn-graql/src/main/java/ai/grakn/graql/internal/query/runner/TinkerComputeQueryRunner.java

```diff
@@ -0,0 +1,683 @@
+import ai.grakn.graql.analytics.CorenessQuery;
+import ai.grakn.graql.analytics.CountQuery;
+import ai.grakn.graql.analytics.DegreeQuery;
+import ai.grakn.graql.analytics.KCoreQuery;
+import ai.grakn.graql.analytics.MaxQuery;
+import ai.grakn.graql.analytics.MeanQuery;
+import ai.grakn.graql.analytics.MedianQuery;
+import ai.grakn.graql.analytics.MinQuery;
+import ai.grakn.graql.analytics.PathQuery;
+import ai.grakn.graql.analytics.PathsQuery;
+import ai.grakn.graql.analytics.StdQuery;
+import ai.grakn.graql.analytics.SumQuery;
+import ai.grakn.graql.internal.analytics.ClusterMemberMapReduce;
+import ai.grakn.graql.internal.analytics.ClusterSizeMapReduce;
+import ai.grakn.graql.internal.analytics.ConnectedComponentVertexProgram;
+import ai.grakn.graql.internal.analytics.ConnectedComponentsVertexProgram;
+import ai.grakn.graql.internal.analytics.CorenessVertexProgram;
+import ai.grakn.graql.internal.analytics.CountMapReduceWithAttribute;
+import ai.grakn.graql.internal.analytics.CountVertexProgram;
+import ai.grakn.graql.internal.analytics.DegreeDistributionMapReduce;
+import ai.grakn.graql.internal.analytics.DegreeStatisticsVertexProgram;
+import ai.grakn.graql.internal.analytics.DegreeVertexProgram;
+import ai.grakn.graql.internal.analytics.GraknMapReduce;
+import ai.grakn.graql.internal.analytics.GraknVertexProgram;
+import ai.grakn.graql.internal.analytics.KCoreVertexProgram;
+import ai.grakn.graql.internal.analytics.MaxMapReduce;
+import ai.grakn.graql.internal.analytics.MeanMapReduce;
+import ai.grakn.graql.internal.analytics.MedianVertexProgram;
+import ai.grakn.graql.internal.analytics.MinMapReduce;
+import ai.grakn.graql.internal.analytics.NoResultException;
+import ai.grakn.graql.internal.analytics.ShortestPathVertexProgram;
+import ai.grakn.graql.internal.analytics.StdMapReduce;
+import ai.grakn.graql.internal.analytics.SumMapReduce;
+import ai.grakn.graql.internal.analytics.Utility;
+import ai.grakn.util.CommonUtil;
+import ai.grakn.util.Schema;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Sets;
+import org.apache.tinkerpop.gremlin.process.computer.ComputerResult;
+import org.apache.tinkerpop.gremlin.process.computer.MapReduce;
+import org.apache.tinkerpop.gremlin.process.computer.Memory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.io.Serializable;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ *
+ *
+ * @author Felix Chapman
+ */
+public class TinkerComputeQueryRunner {
```

Comment:
Big class is big, maybe we can decompose this a bit? Also javadocs could be better.

Reply:
I agree. I split it a bit, but I'll try to split it further.